rfc/rfc017-distributed-network-grpc-v2.md

```diff
@@ -45,4 +54,154 @@ As specified by RFC015, the node MUST authenticate the peer's node DID as follow
+The `LC` value MUST equals the highest Lamport Clock value of all transaction references included in the `XOR` calculation.
+If no transactions are present, an all-zero `XOR` and `LC` of 0 is sent.
+A node is free to choose how the `token` is created as long as it's able to satisfy the requirements of §4.2.3.
+It MUST be valid for at least 10 seconds.
+
+### 4.2.2 IBLT response
+
+When a peer receives an `AdvertState` message and the `XOR` differs, it SHOULD respond with a `TransactionSet` message.
+The upper bound of the response IBLT MUST be the equal to `PSIZE*ceil(LC/PSIZE)` where `LC` is the value from the `AdvertState` message and `ceil` is a function that rounds a double up to the nearest integer.
+If `LC` equals zero, `PSIZE` is used.
+This upper bound is exclusive, the lower bound is inclusive and always 0.
+Next to the IBLT data, the `TransactionSet` message MUST contain the original `LC` value as `LC_req`, a new `LC` value indicating the highest LC value from the peer and the `token` value from the `AdvertState` message.
+
+### 4.2.3 Transaction List Query
+
+For every `TransactionSet` message a node receives, it MUST check the `token` value in the message and compare it to its own administration.
+If the `token` doesn't exist or if the `LC_req` value doesn't match, the node MUST ignore the message.
+
+The IBLT in the `TransactionSet` message contains every transaction of the peer in the LC range of `0-LC_req`. The node MUST lookup/compute the IBLT for this range and subtract it from the IBLT of the peer. Deconstruction of the resulting IBLT will list the transaction refs the peer has and the local node misses. 
+
+These transactions MUST be queried by using the `TransactionListQuery` message. It contains all missing transaction references.
+The message MUST contain a `token` that is to be used in the response.
+
+If the deconstruction fails, the local node sends a new `AdvertState` message. The `LC` value MUST be `PSIZE*floor((LC_req-1)/PSIZE)` where `floor` is a function that rounds a double down to the nearest integer. 
+The `XOR` value MUST be calculated over the transaction references in the range starting with 0 and ending with the new `LC` value.
+
+### 4.2.4 Transaction Range Query
+
+If the IBLT from a `TransactionSet` message could be deconstructed, the local node MAY also send a `TransactionRangeQuery` message.
+This is only needed if the `LC` value of the `TransactionSet` message is equal or larger than `PSIZE*ceil(LC_req/PSIZE)`. 
+This means the peer has additional transactions outside the IBLT range.
+The `TransactionRangeQuery` message contains a `begin` and `end` parameter indicating a request for transactions which LC value is between those parameters.
+The `begin` MUST be set to `PSIZE*ceil(LC_req/PSIZE)` and the end MUST be set to `LC`. Both are inclusive.
```

Comment:
kinda confusing that `end` is inclusive here, and non-inclusive in the earlier IBLT range?

Reply:
fixed